src/jit/lowerxarch.cpp

```diff
@@ -3256,10 +3266,10 @@ void Lowering::LowerCmp(GenTreePtr tree)
         }
         else
         {
-            // One of op1 or op2 could be marked as reg optional
-            // to indicate that codgen can still generate code
-            // if one of them is on stack.
-            SetRegOptional(PreferredRegOptionalOperand(tree));
+            // op1 and op2 could be marked as reg optional since
+            // both of them are not contained.
```

Comment:
This would read more clearly as "Both op1 and op2 can be marked as reg optional since neither of them are contained."